File: TP2_6216948_PW/TP2_6216948_PWS/Models/Team.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace TP2_6216948_PWS.Models
{
    public class Team
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Ville { get; set; }

        public string Commanditaire { get; set; }

        //prop nav
    
        [ForeignKey("League")]
        public int LeagueId { get; set; }
        [JsonIgnore]
        public virtual League League { get; set; }




        [ForeignKey("Arena")]
        public int ArenaID { get; set; }
        [JsonIgnore]
        public virtual Arena Arena { get; set; }


        [ForeignKey("DG")]
        public int DGId { get; set; }
        [JsonIgnore]
        public virtual DG? DG { get; set; }

    }
}

File: TP2_6216948_Angular/src/Models/Team.ts
import { Arena } from "./Arena";
import { DG } from "./DG";
import { League } from "./League";

export class Team{


    constructor(
        public id:number,
        public name:string,
        public ville:string,
        public commanditaire:string,

        public leagueId?:number,
        public league?:League,

        public arenaId?:number,
        public arena?:Arena,

        public dgId?:number,
        public dg?:DG,



    ){}

}